BEIMA.Backend.Test/MongoService/MongoConnectorTest.cs

```diff
@@ -281,5 +281,123 @@ public void ConnectorCreated_UpdateNullDeviceType_NullReturned()
             Assert.IsNull(result);
         }
         #endregion
+
+        #region Building Tests
+
+        [Test]
+        public void ConnectorCreated_GetBuildingGivenValidBuildingId_BuildingDocumentReturned()
+        {
+            var mongo = MongoConnector.Instance;
+            //Setup (inserting a document)
+            var doc = new BsonDocument {
+                { "name", "TestBuilding"},
+                { "description", "This is a test" }
+            };
+            var insertResult = mongo.InsertBuilding(doc);
+            Assume.That(insertResult, Is.Not.Null);
+            Assume.That(insertResult, Is.TypeOf(typeof(ObjectId)));
+
+            //Test (retrieve the document)
+            var retrievedDoc = mongo.GetBuilding((ObjectId)doc["_id"]);
+            Assert.IsNotNull(retrievedDoc);
+            Assert.That(retrievedDoc, Is.TypeOf(typeof(BsonDocument)));
+        }
+
+        [Test]
+        public void ConnectorCreated_GetAllBuildings_BuildingDocumentListReturned()
+        {
+            var mongo = MongoConnector.Instance;
+            //Setup (inserting a document)
+            var doc = new BsonDocument {
+                { "name", "TestBuilding"},
+                { "description", "This is a test" }
+            };
+            var insertResult = mongo.InsertBuilding(doc);
+            Assume.That(insertResult, Is.Not.Null);
+            Assume.That(insertResult, Is.TypeOf(typeof(ObjectId)));
+
+            //Test (retrieve all documents)
+            var retrievedDocs = mongo.GetAllBuildings();
+            Assert.IsNotNull(retrievedDocs);
+            Assert.That(retrievedDocs, Is.TypeOf(typeof(List<BsonDocument>)));
+        }
+
+        [Test]
+        public void InsertBuilding_DeleteBuilding_BuildingHasBeenDeleted()
+        {
+            var mongo = MongoConnector.Instance;
+            var doc = new BsonDocument {
+                { "name", "TestBuilding"},
+                { "description", "This is a test" }
+            };
+            //Insert building
+            var insertResult = mongo.InsertBuilding(doc);
+            Assume.That(insertResult, Is.Not.Null);
+            Assume.That(insertResult, Is.TypeOf(typeof(ObjectId)));
+
+            //Delete building
+            bool deleteResult = false;
+            if (insertResult != null)
+            {
+                deleteResult = mongo.DeleteBuilding((ObjectId)insertResult);
+                Assert.IsNull(mongo.GetBuilding((ObjectId)insertResult));
+            }
+            Assert.IsTrue(deleteResult);
```

Comment:
```suggestion
            Assert.That(deleteResult, Is.True);
```
I prefer to avoid Assert.IsTrue if possible for debugging reasons

Reply:
Fixed [af79ed3](https://github.com/cs481-ekh/s22-beima/pull/250/commits/af79ed301195a5757b220dd36a8958b712be8a1d)